wrangler-core/src/main/java/io/cdap/directives/validation/ValidateStandard.java

```diff
@@ -0,0 +1,213 @@
+        String.format(
+            "[one of: %s]", String.join(", ", standardsManifest.getStandards().keySet())));
+
+    return builder.build();
+  }
+
+  @Override
+  public void initialize(Arguments args) throws DirectiveParseException {
+    column = ((ColumnName) args.value(COLUMN)).value();
+    String spec = ((Identifier) args.value(STANDARD_SPEC)).value();
+
+    if (spec.equals("")) {
+      throw new DirectiveParseException("No standard specified to validate against");
+    }
+    if (standardsManifest == null) {
+      throw new DirectiveParseException(
+          "Standards manifest was not loaded. Please check logs for information");
+    }
+
+    Map<String, Standard> availableSpecs = standardsManifest.getStandards();
+    if (!availableSpecs.containsKey(spec)) {
+      throw new DirectiveParseException(
+          String.format(
+              "Unknown standard %s. Known values are %s",
+              spec, String.join(", ", standardsManifest.getStandards().keySet())));
+    }
+
+    Standard standard = availableSpecs.get(spec);
+    schema =
+        Paths.get(SCHEMAS_RESOURCE_PATH, String.format("%s.%s", spec, standard.getFormat()))
+            .toString();
+
+    if (!schemaToConformer.containsKey(schema)) {
+      if (!FORMAT_TO_FACTORY.containsKey(standard.getFormat())) {
+        throw new DirectiveParseException(
+            String.format("No validator for format %s", standard.getFormat()));
+      }
+
+      InputStream schemaStream;
+      try {
+        schemaStream = readResource(schema);
+      } catch (IOException e) {
+        throw new DirectiveParseException(
+            String.format("Unable to read standard schema: %s", e.getMessage()), e);
+      }
+
+      Conformer<JsonObject> conformer =
+          FORMAT_TO_FACTORY.get(standard.getFormat()).setSchemaStream(schemaStream).build();
+      conformer.initialize();
+      schemaToConformer.put(schema, conformer);
+    }
+  }
+
+  @Override
+  public List<Row> execute(List<Row> rows, ExecutorContext context)
+      throws DirectiveExecutionException, ErrorRowException, ReportErrorAndProceed {
+    for (Row row : rows) {
+      int idx = row.find(column);
+      if (idx != -1) {
+        Object object = row.getValue(idx);
+        if (object instanceof JsonObject) {
+          Conformer<JsonObject> conformer = schemaToConformer.get(schema);
+          if (conformer == null) {
+            throw new DirectiveExecutionException(
+                "Directive was not initialized for schema " + schema);
+          }
+
+          List<ConformanceIssue> conformanceIssues =
+              conformer.checkConformance((JsonObject) object);
+          if (conformanceIssues.size() > 0) {
+            throw new ErrorRowException(
+                conformanceIssues.stream()
+                    .map(ConformanceIssue::toString)
+                    .collect(Collectors.joining("; ")),
+                1, true);
+          }
+        } else if (object != null) {
```

Comment:
nit: similar comment here, it's more readable to do the check first and remove a layer of nesting:

```
if (object == null) {
  continue;
}
if (!(object instanceof JsonObject)) {
  throw ...
}
...
```

Reply:
Done.